addons/sms/models/sms_sms.py

```diff
@@ -83,6 +83,12 @@ def _compute_sms_tracker_id(self):
         for sms in self.filtered(lambda s: s.uuid in tracker_ids_by_sms_uuid):
             sms.sms_tracker_id = tracker_ids_by_sms_uuid[sms.uuid]
 
+    @api.depends('partner_id')
+    def _compute_number(self):
+        target_number = self.env['res.partner']._phone_get_number_fields()
+        for sms in self:
+            sms.number = sms.partner_id[target_number[0]]
+
```

Comment:
I was saying more in the lines of

```
for sms in self:
     for field in self.env['res.partner']._phone_get_number_fields():
          if sms.partner_id[field]:
               sms.number = sms.partner_id[field]
               break
```
So check all valid sms fields, and pick the first non-empty one (we don't care if it's invalid, people shouldn't be using this menu seriously anyway and they'll get an error telling them it's invalid)